src/libasr/ASR.asdl

```diff
@@ -311,6 +311,8 @@ expr
     | DictLen(expr arg, ttype type, expr? value)
 
     | Var(symbol v)
+    | AddressOf(symbol v, ttype type)
+    | DereferencePointer(symbol v, ttype type)
```

Comment:
If we introduce a node like this one, it should be consistently used everywhere. I think it might simplify the LLVM backend with all the loads.

Reply:
I think a compile time value is needed. Furthermore, isn't this the same as Var? Var currently automatically dereferences.

Reply:
> I think a compile time value is needed.

I don't think so. Pointer is an address, available at runtime. So we can only know what's in the address when we run the code. Consider the following example,

```zsh
#include <iostream>

int main() {

    int p = 10;
    const int* ptr = &p;

    p = 20;

    std::cout << *ptr << std::endl;

    return 0;
}
```

```zsh
(lc) 14:43:21:~/lc_project/lc % ./a.out             
20
```